src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Helpers;
+using Microsoft.DotNet.Darc.Options.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Operations.VirtualMonoRepo;
+
+internal class VmrDiffOperation(
+    VmrDiffOptions options,
+    IProcessManager processManager,
+    IFileSystem fileSystem,
+    IGitRepoFactory gitRepoFactory,
+    IVersionDetailsParser versionDetailsParser,
+    IVmrPatchHandler patchHandler,
+    ISourceMappingParser sourceMappingParser) : Operation
+{
+    private const string GitDirectory = ".git";
+    private readonly static string GitSparseCheckoutFile = Path.Combine(GitDirectory, "info", "sparse-checkout");
+    private const string HttpsPrefix = "https://";
+
+    public override async Task<int> ExecuteAsync()
+    {
+        (DiffRepo repo1, DiffRepo repo2) = await ParseInput();
+
+        string tmpPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        try
+        {
+            string tmpProductRepo, tmpVmrProductRepo;
```

Comment:
Can we just do:
```
(string tmpProductRepo, tmpVmrProductRepo) = repo1.IsVmr ? (repo2, repo1) : ...;
```

And then not repeat the code below?

Reply:
ah thanks, didn't notice it was exactly the same code